api/src/main/java/com/spotify/ffwd/model/v2/Metric.java

```diff
@@ -0,0 +1,81 @@
+/*-
+ * -\-\-
+ * FastForward API
+ * --
+ * Copyright (C) 2016 - 2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.ffwd.model.v2;
+
+import com.google.common.base.Charsets;
+import com.google.common.hash.HashFunction;
+import com.google.common.hash.Hasher;
+import com.google.common.hash.Hashing;
+import com.spotify.ffwd.model.Metrics;
+import java.util.Date;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+
+@Data
+@EqualsAndHashCode(of = {"key", "tags"})
+public class Metric implements Metrics {
```

Comment:
I'm hesitant to go ahead with this change. Why can't we use original `Metric` and `Batch` objects and use new `Value` type instead of `double`? The `Metric` and `Batch` objects are internal to FFWD and I don't believe we use them anywhere. This then will introduce only Value object and change the Metric and Batch as they are.

Reply:
Technically, It is not internal. Http Reporter and non java-clients  must conform to this format, right ? 
 I think this is risk free unless there is a parity between instrumentation side and ffwd agent.

Reply:
This Metric object is used internally only.

Reply:
Let's chat offline about this. Metric definition can be confusing. Http client is using this format even if the object is not shared.